synapse/lib/oauth.py

```diff
@@ -55,65 +59,54 @@
     'required': ['access_token', 'expires_in'],
 })
 
-KEY_PART_LEN = 32
-REFRESH_WINDOW = 0.5  # refresh in REFRESH_WINDOW * expires_in
-
-class OAuthManager(s_nexus.Pusher):
+def normOAuthTokenData(issued_at, data):
     '''
-    Organize and execute OAuth token refreshes.
+    Normalize timestamps to be in epoch millis and set expires_at/refresh_at.
+    '''
+    reqValidTokenResponse(data)
+    expires_in = data['expires_in']
+    return {
+        'access_token': data['access_token'],
+        'expires_in': expires_in,
+        'expires_at': issued_at + expires_in * 1000,
+        'refresh_at': issued_at + (expires_in * REFRESH_WINDOW) * 1000,
+        'refresh_token': data.get('refresh_token'),
+    }
+
+class OAuthMixin(s_nexus.Pusher):
+    '''
+    Mixin for Cells to organize and execute OAuth token refreshes.
     '''
 
-    async def __anit__(self, cell):
-        await s_nexus.Pusher.__anit__(self, 'oauth', nexsroot=cell.nexsroot)
-
-        self.cell = cell
-
-        slab = self.cell.slab
-        self.clients = s_lmdbslab.SlabDict(slab, db=slab.initdb('oauth:v2:clients'))     # key=<provideriden><useriden>
-        self.providers = s_lmdbslab.SlabDict(slab, db=slab.initdb('oauth:v2:providers')) # key=<provideriden>
+    async def _initOAuthManager(self):
 
-        self.ssl = None
-        cadir = self.cell.conf.get('tls:ca:dir')
-        if cadir is not None:
-            self.ssl = s_common.getSslCtx(cadir)
+        slab = self.slab
+        self.oauth_clients = s_lmdbslab.SlabDict(slab, db=slab.initdb('oauth:v2:clients'))     # key=<provider><user>
```

Comment:
is there a reason to leave any of these mixin variables without a `_` identifier? All other interaction with this should be through the APIs it exposes.